### PR TITLE
refactor: handle multiple unlock types in `getInputsPreExpandedConfig`

### LINKED ISSUE #1104: [Bug]: Explorer crashes when viewing Token from L2
### Page gets blank

Page gets white after i view a transaction from a token sent from L2 to L1 with the toolkit https://evm-toolkit.evm.shimmer.network/
Error

this is my transaction: https://explorer.shimmer.network/shimmer/transaction/0x02b8488d0cd5ff39d8ade27a73c11e85bbc46171e4479d60f194869d7ebdc7d1

### Expected behaviour

Explorer should show the tx

### Actual behaviour

Page gets white after i view a transaction from a token sent from L2 to L1 with the toolkit https://evm-toolkit.evm.shimmer.network/

### Can the issue reliably be reproduced?

Yes

### Steps to reproduce the issue

1. Send a token, not SMR with the toolkit to L1 https://evm-toolkit.evm.shimmer.network/
2. View the transaction from firefly or so

### Errors

_No response_

### Duplicate declaration

- [X] I have searched the issues tracker for this issue and there is none

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by VmMad**
# Description of change

Fixes #1104 

## Type of change

- Bug fix (a non-breaking change which fixes an issue)

## How the change has been tested

`shimmer/transaction/0x02b8488d0cd5ff39d8ade27a73c11e85bbc46171e4479d60f194869d7ebdc7d1`

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: client/src/helpers/stardust/preExpandedConfig.ts [hunk 1]
```diff
@@ -45,7 +45,8 @@ export function getInputsPreExpandedConfig(inputs: IInput[], unlocks: Unlock[],
             if (input?.output?.output && "unlockConditions" in input.output.output) {
                 const commmonOutput = input.output.output as unknown as CommonOutput;
                 let unlock = unlocks[idx];
-                if (unlock.type === UnlockType.Reference) {
+                const referenceUnlocks = [UnlockType.Reference, UnlockType.Alias, UnlockType.Nft];
```
Comment (msarcev) [inline client/src/helpers/stardust/preExpandedConfig.ts:48]: There are more `UnlockType`s that this though ? Shouldn't we cover all of them.
@begonaalvarezd Apart from that, now that i red it better, not sure that this logic is solid, cause "referenceUnlocks" can be transitive.
Check how we do it in `client/src/helpers/stardust/transactionsHelper.ts` (`getInputsAndOutputs`).

Comment (begonaalvarezd) [reply client/src/helpers/stardust/preExpandedConfig.ts:48]: image

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
unlock references can be transitive, so we need to follow the path until we find the signature

**Review by begonaalvarezd — CHANGES_REQUESTED** [review]
there seems to be a bug, if you visit the tx of the original issue `/shimmer/transaction/0x02b8488d0cd5ff39d8ade27a73c11e85bbc46171e4479d60f194869d7ebdc7d1` there is a basic output that doesnt expand its address unlock
image

**Comment by VmMad** [comment]
> there seems to be a bug, if you visit the tx of the original issue `/shimmer/transaction/0x02b8488d0cd5ff39d8ade27a73c11e85bbc46171e4479d60f194869d7ebdc7d1` there is a basic output that doesnt expand its address unlock image

Issue reference: #1121

**Review by begonaalvarezd — APPROVED** [review]

**Review by msarcev — APPROVED** [review]

**Pull request merged**